substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ClassLoaderSupportFeature.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2021, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.hosted;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+import org.graalvm.compiler.serviceprovider.JavaVersionUtil;
+import org.graalvm.nativeimage.ImageSingletons;
+import org.graalvm.nativeimage.hosted.Feature;
+
+import com.oracle.svm.core.ClassLoaderSupport;
+import com.oracle.svm.core.annotate.AutomaticFeature;
+import com.oracle.svm.core.util.VMError;
+
+@AutomaticFeature
+public class ClassLoaderSupportFeature implements Feature {
+    @Override
+    public void afterRegistration(AfterRegistrationAccess a) {
+        VMError.guarantee(JavaVersionUtil.JAVA_SPEC <= 8);
```

Comment:
So looks like you are still relying on the multi-version-jar mechanism to distinguish JDK versions. I thought we agreed to not use that mechanism, ans instead do explicit version checks  in the feature before registration.

Reply:
If I understand correctly the idea is to remove the jdk11 version of `com.oracle.svm.hosted.ClassLoaderSupportFeature` and to let `src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ClassLoaderSupportFeature.java` instead register 
`ImageSingletons.add(ClassLoaderSupport.class, new ClassLoaderSupportImpl11....)` for `JavaVersionUtil.JAVA_SPEC > 8` and `ImageSingletons.add(ClassLoaderSupport.class, new ClassLoaderSupportImpl8....)` otherwhise, right?

This should work if I add a dummy `ClassLoaderSupportImpl11` for 8 so that `src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ClassLoaderSupportFeature.java` compiles.

Reply:
Use same scheme as `RecordSupport`

Reply:
Fixed: https://github.com/oracle/graal/pull/3446/commits/89bb920793c0fdd774e467bb8729dc9f3c496bad